_data/data.js

```diff
@@ -244,6 +274,12 @@ const downloadImage = async function (url, file) {
     writer.on('finish', () => {
       console.timeEnd(label);
       console.log(`Image downloaded: ${path}`);
+
+      downloadedImagesURLs.set(url, {
+        path,
+        virtualPath,
+      });
```

Comment:
Putting a downloaded image to the map here is prone to race condition. If A and B are pointing to the same URL but one starts downloading before the other finishes downloading, what would happen? Any idea on how to fix this?